astropy/nddata/utils.py

```diff
@@ -11,6 +11,11 @@
            'overlap_slices']
 
 
+class NoOverlapError(ValueError):
```

Comment:
Would it be more logical to base this on `IndexError`? (since that is what is given with `np.array([0])[1]`)